test/money.html

```diff
@@ -46,58 +47,70 @@
+				assert.equal(amountEquals({ amount, currency: 'EUR' }, { amount, currency: 'EUR' }), true);
+				assert.equal(amountEquals({ amount: 1.01, currency: 'EUR' }, { amount: 1.009, currency: 'EUR' }, 2), true);
+				assert.equal(amountEquals({ amount: 1.01, currency: 'EUR' }, { amount: 1.009, currency: 'EUR' }, 3), false);
+				assert.equal(amountEquals({ amount: 1.100, currency: 'EUR' }, { amount: 1.1001, currency: 'EUR' }, 2), true);
+				assert.equal(amountEquals({ amount: -1.1001, currency: 'EUR' }, { amount: -1.100, currency: 'EUR' }, 2), true);
+				assert.equal(amountEquals({ amount, currency: 'EUR' }, { amount, currency: 'EUR' }, 4), true);
+			});
 
-		test('renderAmount ', () => {
-			assert.equal(renderAmount({ amount }), undefined);
-			assert.isObject(locAmount, 'testamount is configured for locale');
-			testCurrencies.forEach(currency => {
-				const output = locAmount[alt ? currency + '2' : currency] || locAmount[currency];
-				assert.isString(output, `testcurrency is configured for testamount - ${alt} ${isEdge} ${isSafari}`);
-				assert.equal(renderAmount({ amount, currency }), output);
+			test('isAmount ', () => {
+				assert.equal(isAmount(null), false);
+				assert.equal(isAmount(false), false);
+				assert.equal(isAmount(100), false);
+				assert.equal(isAmount({ amount }), false);
+				assert.equal(isAmount({ currency: 'EUR' }), false);
+				assert.equal(isAmount({ amount, currency: 'EUR' }), true);
+				assert.equal(isAmount({ amount: amount.toString(), currency: 'EUR' }), false);
+				assert.equal(isAmount({ amount: null, currency: 'USD' }), false);
+				assert.equal(isAmount({ amount, currency: null }), false);
 			});
-			assert.equal(renderAmount(null), undefined);
-			assert.equal(renderAmount(), undefined);
-			assert.equal(renderAmount({ amount, currency: 'USD' }), renderMoney({ amount, currency: 'USD' }));
-		});
 
-		test('renderNumberAmount ', () => {
-			assert.equal(renderNumberAmount({ amount, currency: 'USD' }), locAmount.number);
+			test('renderAmount ', () => {
+				assert.equal(renderAmount({ amount }, locale), undefined);
+				assert.isObject(locAmount, 'testamount is configured for locale');
+
+				testCurrencies.forEach(currency => {
+					const output = locAmount[alt ? currency + '2' : currency] || locAmount[currency];
+					assert.isString(output, `testcurrency is configured for testamount - ${alt} ${isEdge} ${isSafari}`);
+					assert.equal(renderAmount({ amount, currency}, locale), output);
+				});
+				assert.equal(renderAmount(null), undefined);
+				assert.equal(renderAmount(), undefined);
+				assert.equal(
+					renderAmount({
+						amount, currency: 'USD'
+					}, locale),
+					renderMoney({
+						amount, currency: 'USD'
+					}, locale)
+				);
+			});
+
+			test('renderNumberAmount ', () => {
+				assert.equal(renderNumberAmount({ amount, currency: 'USD' }, locale), locAmount.number);
+			});
```

Comment:
You should also add a couple of tests for calling `renderAmount` and `renderNumberAmount` without the `locale` parameter, to make sure it does not error out if `locale` is missing (which is the case in all existing frontend code), but you should test the result in a generic fashion, i.e. if the result is a string, if it has a min length, etc, otherwise we will again run into errors caused by the different saucelabs locale settings.

Reply:
Added some tests to check type and minimum length for `renderAmount` and `renderNumberAmount`.